language/tools/move-cli/src/base/commands/check.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use move_lang::{self, shared::Flags};
+
+/// Compile the user modules in `src` and the script in `script_file`
+pub fn check(
+    interface_files: &[String],
+    republish: bool,
```

Comment:
Do we want/need this flag for a non sandbox command? It might be that we do, but should just name it something else...

Reply:
I think keeping it makes sense so it can be reused in the sandbox command as well. For the base command however, we'll pass in a default of `false` here. 

I'll change this to what the compiler calls this (`sources_shadow_deps`) if that sounds good to you.